### PR TITLE
fix(shwap/bitswap): close acccessor in Blockstore

### PR DESCRIPTION
**Pull request by Wondertan**
Blockstore must cleanup the accesor

### CODE REVIEW

File: share/shwap/p2p/bitswap/block_fetch_test.go [hunk 1]
```diff
@@ -117,7 +117,7 @@ func TestFetch_Duplicates(t *testing.T) {
 func newExchangeOverEDS(ctx context.Context, t *testing.T, rsmt2d *rsmt2d.ExtendedDataSquare) exchange.SessionExchange {
 	bstore := &Blockstore{
 		Getter: testAccessorGetter{
-			Accessor: eds.Rsmt2D{ExtendedDataSquare: rsmt2d},
+			AccessorCloser: eds.WithCloser(eds.Rsmt2D{ExtendedDataSquare: rsmt2d}, nopCloser{}),
```
Comment (Wondertan) [inline share/shwap/p2p/bitswap/block_fetch_test.go:120]: That is exactly why I don't like the `eds.WithCloser` approach @walldiss

**Comment by walldiss** [comment]
Unfortunately would need rebase after merge of streaming.

File: share/shwap/p2p/bitswap/block_store.go [hunk 2]
```diff
@@ -38,6 +38,11 @@ func (b *Blockstore) getBlock(ctx context.Context, cid cid.Cid) (blocks.Block, e
 	if err != nil {
 		return nil, fmt.Errorf("getting EDS Accessor for height %v: %w", blk.Height(), err)
 	}
+	defer func() {
+		if err := eds.Close(); err != nil {
+			log.Warnf("failed to close EDS accessor for height %v: %s", blk.Height(), err)
+		}
+	}()
```
Comment (walldiss) [inline share/shwap/p2p/bitswap/block_store.go:45]: We now have handy utils.CloseAndLog for that, as it becomes common code accross the repo

Comment (Wondertan) [reply share/shwap/p2p/bitswap/block_store.go:45]: Its not merged to shwap yet and i dont find it useful honestly. I am gonna keep it as it is here

Comment (Wondertan) [reply share/shwap/p2p/bitswap/block_store.go:45]: The log message should contain the context of where the warn/error happens. Using this util will remove any context

Comment (walldiss) [reply share/shwap/p2p/bitswap/block_store.go:45]: It takes logger as first argument, so you can pass as much additional context information as you want within loggier fields.

Comment (walldiss) [reply share/shwap/p2p/bitswap/block_store.go:45]: This is optional, so  would defer to you to decide.

**Review by walldiss — APPROVED** [review]

File: share/shwap/p2p/bitswap/block_store.go (hunk 2 above)
Comment (Wondertan) [reply share/shwap/p2p/bitswap/block_store.go:45]: i dont see any value so gonna keep the way it is

**Pull request merged**